<?xml version="1.0" encoding="UTF-8"?>
<pd:ProcessDefinition xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:pfx="http://www.tibco.com/pe/DeployedVarsType" xmlns:pd="http://xmlns.tibco.com/bw/process/2003" xmlns:pfx2="http://xmlns.heb.com/ei/error" xmlns:BW="java://com.tibco.pe.core.JavaCustomXPATHFunctions" xmlns:ns="http://www.tibco.com/pe/EngineTypes" xmlns:pfx3="http://www.tibco.com/ns/no_namespace_schema_location/Shared Resources/Schema Definitions/Common/Exception.xsd" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:ns2="http://www.tibco.com/namespaces/tnt/plugins/jms" xmlns:tib="http://www.tibco.com/bw/xslt/custom-functions" xmlns:ns1="www.tibco.com/plugin/Sleep" xmlns:ns3="http://www.tibco.com/pe/GenerateErrorActivity/InputSchema">
    <xsd:import namespace="http://www.tibco.com/ns/no_namespace_schema_location/Shared Resources/Schema Definitions/Common/Exception.xsd" schemaLocation="/Shared Resources/Schema Definitions/Common/Exception.xsd"/>
    <pd:name>PsCustomerDepositsToHEBCard Processes/Starter Process/CustomerDeposits_Starter.process</pd:name>
    <pd:startName>onStartup</pd:startName>
    <pd:startX>0</pd:startX>
    <pd:startY>0</pd:startY>
    <pd:returnBindings/>
    <pd:starter name="onStartup">
        <pd:type>com.tibco.pe.core.OnStartupEventSource</pd:type>
        <pd:resourceType>ae.activities.onStartup</pd:resourceType>
        <pd:x>80</pd:x>
        <pd:y>219</pd:y>
        <config/>
        <pd:inputBindings/>
    </pd:starter>
    <pd:endName>End</pd:endName>
    <pd:endX>903</pd:endX>
    <pd:endY>218</pd:endY>
    <pd:errorSchemas>
        <Exception ref="pfx3:Exception"/>
    </pd:errorSchemas>
    <pd:processVariables/>
    <pd:targetNamespace>http://xmlns.example.com/1448660840157</pd:targetNamespace>
    <pd:group name="Retry Group on Error">
        <pd:type>com.tibco.pe.core.LoopGroup</pd:type>
        <pd:resourceType>ae.process.group</pd:resourceType>
        <pd:x>262</pd:x>
        <pd:y>23</pd:y>
        <pd:width>348</pd:width>
        <pd:height>378</pd:height>
        <pd:collapsedWidth>57</pd:collapsedWidth>
        <pd:collapsedHeight>70</pd:collapsedHeight>
        <pd:groupVariables/>
        <pd:joinToggle>inherit</pd:joinToggle>
        <config>
            <pd:groupType>errorLoop</pd:groupType>
            <pd:serializable>false</pd:serializable>
            <pd:indexSlot>count</pd:indexSlot>
            <pd:errorCondition>$count &gt;= $_globalVariables/pfx:GlobalVariables/RetryConfig/GLB_MAX_RETRIES</pd:errorCondition>
            <pd:suspendAfterErrorRetry>true</pd:suspendAfterErrorRetry>
        </config>
        <pd:inputBindings/>
        <pd:expanded>true</pd:expanded>
        <pd:transition>
            <pd:from>start</pd:from>
            <pd:to>Call_Process_Main</pd:to>
            <pd:lineType>Multiple Bends</pd:lineType>
            <pd:lineColor>-16777216</pd:lineColor>
            <pd:conditionType>always</pd:conditionType>
        </pd:transition>
        <pd:transition>
            <pd:from>Call_Process_Main</pd:from>
            <pd:to>end</pd:to>
            <pd:lineType>One Bend</pd:lineType>
            <pd:lineColor>-16777216</pd:lineColor>
            <pd:conditionType>always</pd:conditionType>
        </pd:transition>
        <pd:transition>
            <pd:from>Validate Error</pd:from>
            <pd:to>end</pd:to>
            <pd:xpathDescription>Is Error Data</pd:xpathDescription>
            <pd:lineType>Default</pd:lineType>
            <pd:lineColor>-16777216</pd:lineColor>
            <pd:conditionType>xpath</pd:conditionType>
            <pd:xpath>$Validate-Error/ErrorCode='true'</pd:xpath>
        </pd:transition>
        <pd:transition>
            <pd:from>Catch</pd:from>
            <pd:to>Validate Error</pd:to>
            <pd:xpathDescription>is Error</pd:xpathDescription>
            <pd:lineType>One Bend</pd:lineType>
            <pd:lineColor>-3407872</pd:lineColor>
            <pd:conditionType>always</pd:conditionType>
        </pd:transition>
        <pd:transition>
            <pd:from>Sleep</pd:from>
            <pd:to>ErrorProcess</pd:to>
            <pd:xpathDescription>Count =3</pd:xpathDescription>
            <pd:lineType>Default</pd:lineType>
            <pd:lineColor>-16777216</pd:lineColor>
            <pd:conditionType>xpath</pd:conditionType>
            <pd:xpath>$count = $_globalVariables/pfx:GlobalVariables/RetryConfig/GLB_MAX_RETRIES</pd:xpath>
        </pd:transition>
        <pd:transition>
            <pd:from>Sleep</pd:from>
            <pd:to>Generate-Error-1</pd:to>
            <pd:xpathDescription>No</pd:xpathDescription>
            <pd:lineType>Default</pd:lineType>
            <pd:lineColor>-16777216</pd:lineColor>
            <pd:conditionType>otherwise</pd:conditionType>
        </pd:transition>
        <pd:transition>
            <pd:from>Validate Error</pd:from>
            <pd:to>Sleep</pd:to>
            <pd:xpathDescription>Is Technical Error</pd:xpathDescription>
            <pd:lineType>Default</pd:lineType>
            <pd:lineColor>-16777216</pd:lineColor>
            <pd:conditionType>otherwise</pd:conditionType>
        </pd:transition>
        <pd:activity name="Call_Process_Main">
            <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
            <pd:resourceType>ae.process.subprocess</pd:resourceType>
            <pd:x>344</pd:x>
            <pd:y>88</pd:y>
            <config>
                <processName>/PsCustomerDepositsToHEBCard Processes/Main Process/Send_CustomerDeposits_Main.process</processName>
            </config>
            <pd:inputBindings>
                <root/>
            </pd:inputBindings>
        </pd:activity>
        <pd:activity name="Sleep">
            <pd:type>com.tibco.plugin.timer.SleepActivity</pd:type>
            <pd:resourceType>ae.activities.sleep</pd:resourceType>
            <pd:x>442</pd:x>
            <pd:y>331</pd:y>
            <config/>
            <pd:inputBindings>
                <ns1:SleepInputSchema>
                    <IntervalInMillisec>
                        <xsl:value-of select="$_globalVariables/pfx:GlobalVariables/RetryConfig/GBL_RETRY_DELAY"/>
                    </IntervalInMillisec>
                </ns1:SleepInputSchema>
            </pd:inputBindings>
        </pd:activity>
        <pd:activity name="ErrorProcess">
            <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
            <pd:resourceType>ae.process.subprocess</pd:resourceType>
            <pd:x>540</pd:x>
            <pd:y>330</pd:y>
            <config>
                <processName>/Utility Services/Exception Handling/ErrorProcess.process</processName>
            </config>
            <pd:inputBindings>
                <Exception>
                    <Domain>
                        <xsl:value-of select="$_globalVariables/pfx:GlobalVariables/Domain"/>
                    </Domain>
                    <Deployment>
                        <xsl:value-of select="$_globalVariables/pfx:GlobalVariables/Deployment"/>
                    </Deployment>
                    <Project>
                        <xsl:value-of select="$_processContext/ns:ProcessContext/ProjectName"/>
                    </Project>
                    <EngineName>
                        <xsl:value-of select="$_processContext/ns:ProcessContext/EngineName"/>
                    </EngineName>
                    <TimeStamp>
                        <xsl:value-of select="current-dateTime()"/>
                    </TimeStamp>
                    <xsl:if test="$_error/ns:ErrorReport/MsgCode">
                        <Code>
                            <xsl:value-of select="$_error/ns:ErrorReport/MsgCode"/>
                        </Code>
                    </xsl:if>
                    <Message>
                        <xsl:value-of select="$_error/ns:ErrorReport/Msg"/>
                    </Message>
                </Exception>
            </pd:inputBindings>
        </pd:activity>
        <pd:activity name="Catch">
            <pd:type>com.tibco.pe.core.CatchActivity</pd:type>
            <pd:resourceType>ae.activities.catch</pd:resourceType>
            <pd:x>360</pd:x>
            <pd:y>167</pd:y>
            <pd:handler>true</pd:handler>
            <config>
                <catchAll>true</catchAll>
            </config>
            <pd:inputBindings/>
        </pd:activity>
        <pd:activity name="Generate-Error-1">
            <pd:type>com.tibco.pe.core.GenerateErrorActivity</pd:type>
            <pd:resourceType>ae.activities.throw</pd:resourceType>
            <pd:x>348</pd:x>
            <pd:y>332</pd:y>
            <config>
                <faultName/>
            </config>
            <pd:inputBindings>
                <ns3:ActivityInput>
                    <xsl:if test="$_error/ns:ErrorReport/Msg">
                        <message>
                            <xsl:value-of select="$_error/ns:ErrorReport/Msg"/>
                        </message>
                    </xsl:if>
                    <xsl:if test="$_error/ns:ErrorReport/MsgCode">
                        <messageCode>
                            <xsl:value-of select="$_error/ns:ErrorReport/MsgCode"/>
                        </messageCode>
                    </xsl:if>
                </ns3:ActivityInput>
            </pd:inputBindings>
        </pd:activity>
        <pd:activity name="Validate Error">
            <pd:type>com.tibco.plugin.mapper.MapperActivity</pd:type>
            <pd:resourceType>ae.activities.MapperActivity</pd:resourceType>
            <pd:x>439</pd:x>
            <pd:y>222</pd:y>
            <config>
                <element>
                    <xsd:element name="ErrorCode" type="xsd:string"/>
                </element>
            </config>
            <pd:inputBindings>
                <ErrorCode>
                    <xsl:value-of select="contains($_globalVariables/pfx:GlobalVariables/ExceptionHandling/GLB_CONDITION_FOR_DATA_ERROR, $_error/ns:ErrorReport/MsgCode)"/>
                </ErrorCode>
            </pd:inputBindings>
        </pd:activity>
    </pd:group>
    <pd:activity name="Generate Error">
        <pd:type>com.tibco.pe.core.GenerateErrorActivity</pd:type>
        <pd:resourceType>ae.activities.throw</pd:resourceType>
        <pd:x>439</pd:x>
        <pd:y>473</pd:y>
        <config>
            <faultName>Exception</faultName>
        </config>
        <pd:inputBindings>
            <ns3:ActivityInput>
                <xsl:if test="$_error/ns:ErrorReport/Msg">
                    <message>
                        <xsl:value-of select="$_error/ns:ErrorReport/Msg"/>
                    </message>
                </xsl:if>
                <xsl:if test="$_error/ns:ErrorReport/MsgCode">
                    <messageCode>
                        <xsl:value-of select="$_error/ns:ErrorReport/MsgCode"/>
                    </messageCode>
                </xsl:if>
                <data>
                    <Exception>
                        <Domain>
                            <xsl:value-of select="$_globalVariables/pfx:GlobalVariables/Domain"/>
                        </Domain>
                        <Deployment>
                            <xsl:value-of select="$_globalVariables/pfx:GlobalVariables/Deployment"/>
                        </Deployment>
                        <Project>
                            <xsl:value-of select="$_processContext/ns:ProcessContext/ProjectName"/>
                        </Project>
                        <EngineName>
                            <xsl:value-of select="$_processContext/ns:ProcessContext/EngineName"/>
                        </EngineName>
                        <TimeStamp>
                            <xsl:value-of select="current-dateTime()"/>
                        </TimeStamp>
                        <xsl:if test="$_error/ns:ErrorReport/MsgCode">
                            <Code>
                                <xsl:value-of select="$_error/ns:ErrorReport/MsgCode"/>
                            </Code>
                        </xsl:if>
                        <Message>
                            <xsl:value-of select="$_error/ns:ErrorReport/Msg"/>
                        </Message>
                    </Exception>
                </data>
            </ns3:ActivityInput>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Catch-1">
        <pd:type>com.tibco.pe.core.CatchActivity</pd:type>
        <pd:resourceType>ae.activities.catch</pd:resourceType>
        <pd:x>797</pd:x>
        <pd:y>458</pd:y>
        <pd:handler>true</pd:handler>
        <config>
            <fault>localname=Exception</fault>
            <catchAll>true</catchAll>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:activity name="ErrorProcess-1">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>901</pd:x>
        <pd:y>458</pd:y>
        <config>
            <processName>/Utility Services/Exception Handling/ErrorProcess.process</processName>
        </config>
        <pd:inputBindings>
            <Exception>
                <Domain>
                    <xsl:value-of select="$_globalVariables/pfx:GlobalVariables/Domain"/>
                </Domain>
                <Deployment>
                    <xsl:value-of select="$_globalVariables/pfx:GlobalVariables/Deployment"/>
                </Deployment>
                <Project>
                    <xsl:value-of select="$_processContext/ns:ProcessContext/ProjectName"/>
                </Project>
                <EngineName>
                    <xsl:value-of select="$_processContext/ns:ProcessContext/EngineName"/>
                </EngineName>
                <TimeStamp>
                    <xsl:value-of select="current-dateTime()"/>
                </TimeStamp>
                <xsl:if test="$_error/ns:ErrorReport/MsgCode">
                    <Code>
                        <xsl:value-of select="$_error/ns:ErrorReport/MsgCode"/>
                    </Code>
                </xsl:if>
                <Message>
                    <xsl:value-of select="$_error/ns:ErrorReport/Msg"/>
                </Message>
            </Exception>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Call Set_EngineService">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>741</pd:x>
        <pd:y>217</pd:y>
        <config>
            <processName>/Utility Services/Custom Functions/Set_EngineService.process</processName>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:activity name="Call-Set_EngineService-1">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>902</pd:x>
        <pd:y>330</pd:y>
        <config>
            <processName>/Utility Services/Custom Functions/Set_EngineService.process</processName>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:transition>
        <pd:from>Retry Group on Error</pd:from>
        <pd:to>Generate Error</pd:to>
        <pd:xpathDescription>Error</pd:xpathDescription>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>otherwise</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Catch-1</pd:from>
        <pd:to>ErrorProcess-1</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>ErrorProcess-1</pd:from>
        <pd:to>Call-Set_EngineService-1</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>onStartup</pd:from>
        <pd:to>Retry Group on Error</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Retry Group on Error</pd:from>
        <pd:to>Call Set_EngineService</pd:to>
        <pd:xpathDescription>No Error</pd:xpathDescription>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>xpath</pd:conditionType>
        <pd:xpath>not(exists($_error))</pd:xpath>
    </pd:transition>
    <pd:transition>
        <pd:from>Call Set_EngineService</pd:from>
        <pd:to>End</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Call-Set_EngineService-1</pd:from>
        <pd:to>End</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
</pd:ProcessDefinition>